compact_proposal/compact_proposal.cls

```diff
@@ -56,23 +56,16 @@
 
 %% Define note functions
 
-% \newcommand*{\myparenthetical}[1]{%
-%   \ifhmode
-%     \unskip
-%     \space
-%   \fi
-%   [#1]%
-%   \@ifnextchar{.}{}{%
-%   \@ifnextchar{,}{}{%
-%   \@ifnextchar{;}{}{%
-%   \@ifnextchar{!}{}{%
-%   \@ifnextchar{?}{}{%
-%   \@ifnextchar{)}{}{%
-%   \@ifnextchar\par{}{%
-%     \space
-%     \ignorespaces
-%   }}}}}}}%
-% }
+% dynamicSpace command is borred from Heiko Oberdiek:
+% https://tex.stackexchange.com/questions/241038/detect-whether-theres-space-before-and-after-macro
+\newcommand{\dynamicSpace}[1]{%
```

Comment:
I like this as a mechanic to get rid of this really annoying hack I came up with. 

6 internet points.